File: zig/collatz-conjecture/collatz_conjecture.zig
pub const ComputationError = error{
    IllegalArgument,
};

pub fn steps(number: usize) anyerror!usize {
    if (number == 0) {
        return ComputationError.IllegalArgument;
    }
    if (number == 1) {
        return 0;
    }
    var number_use: usize = number;
    var n: usize = 0;
    var result: usize = 0;
    while (result != 1) {
        if (number_use % 2 == 0) {
            result = number_use / 2;
        } else {
            result = 3 * number_use + 1;
        }
        number_use = result;
        n += 1;
    }
    return n;
}

File: typescript/collatz-conjecture/collatz-conjecture.ts
export function steps(count: number): number {
  if (count <= 0 || !Number.isInteger(count)) throw new Error('Only positive integers are allowed');
  if (count === 1) return 0;

  let result = 0;
  for (let i = count; i > 1; i = i % 2 === 0 ? i / 2 : i * 3 + 1) {
    result++;
  }

  return result;
}
